parcels/field.py

```diff
@@ -190,6 +190,11 @@ def __init__(self, name, data, lon=None, lat=None, depth=None, time=None, grid=N
         # self.lon, self.lat, self.depth and self.time are not used anymore in parcels.
         # self.grid should be used instead.
         # Those variables are still defined for backwards compatibility with users codes.
+        if np.all(self.grid.lat[:-1] > self.grid.lat[1:]):
+            # lat has to be in ascending order
+            self.grid.lat = np.flip(self.grid.lat, axis=0)
+            # if we flip lat, we need to flip data too
+            self.data = np.flip(self.data, axis=2)
```

Comment:
I would move these five new lines to *above* the three comment lines on self.lon etc, as these five lines are not per se related to the comment on self.lon
In other words, swap lines 190-192 and 193-197

Reply:
I think the lat flip, should not be done in field.__init__ but in grid.__init__

Then, you can do more specific stuff, since what you did only works for lat vectors. But in curvi grids, we have lat arrays and lat arrays.

So, in RectilinearGrids, you should do the lat.flip, plus a variable grid.latFlipped = True
In RectilinearSGrids, you should have
if grid.latFlipped:
  you flip the y dimension of depth (which is axis=-2)

In Curvi grids, you do the same, but you flip also lon arrays

In field.__init__, you have 
if grid.latFlipped:
  you flip the y dimension of depth (which is axis=-2)

(Can you have a negative value for the axis? Because, ydimension is not alwasy the 2 axis, but always the -2. So what you wrote is not general)